commit 2a5641d597bcc4a9ef1e71631be5d4a99b22a31b (refs/changes/14/5214/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2018-12-10T13:19:31-08:00 (10 months ago)
    
    TRITON-1024 cns cnapi timeout too short, updater crashes on timeout

diff --git a/lib/cn-filter.js b/lib/cn-filter.js
index 8a96634..40fb5b2 100644
--- a/lib/cn-filter.js
+++ b/lib/cn-filter.js
@@ -21,8 +21,8 @@ var consts = require('./consts');
 
 var FSM = require('mooremachine').FSM;
 
-var UPDATE_INTERVAL = 30000;
-var CNAPI_TIMEOUT = 10000;
+var UPDATE_INTERVAL = 60000;
+var CNAPI_TIMEOUT = 30000;
 var RETRY_INTERVAL = 5000;
 
 /*
@@ -202,7 +202,6 @@ UpdateCNCacheFSM.prototype.state_update = function (S) {
 		S.gotoState('fire');
 	}));
 	S.timeout(CNAPI_TIMEOUT, function () {
-		creq.cancel();
 		cnf.log.warn('timed out while updating server cache');
 		S.gotoState('retry');
 	});
@@ -240,7 +239,8 @@ function CNFilter(opts) {
 
 	this.client = restify.createJsonClient(utils.getRestifyClientOptions({
 		url: 'http://' + this.config.address,
-		agent: opts.agent
+		agent: opts.agent,
+		retry: false
 	}));
 
 	this.cache = undefined;
